Mathlib/NumberTheory/Ostrowski.lean

```diff
@@ -28,6 +28,52 @@ absolute values on a number field up to equivalence. (TODO)
 ring_norm, ostrowski
 -/
 
+namespace Filter
+
+-- ## Preliminary lemmas on limits
+
+/-- If `a : ℝ` is bounded above by a function `g : ℕ → ℝ` for every `0 < k`
+then it is less or equal than the limit `lim_{k → ∞} g(k)`-/
+lemma le_of_limit_le {a : ℝ} {g : ℕ → ℝ} {l : ℝ} (ha : ∀ (k : ℕ) (_ : 0 < k), a ≤ g k)
+    (hg : Filter.Tendsto g Filter.atTop (nhds l)) : a ≤ l := by
+  apply le_of_tendsto_of_tendsto tendsto_const_nhds hg
+  rw [Filter.EventuallyLE, Filter.eventually_atTop]
+  exact ⟨1, ha⟩
+
+/-- For any `C > 0`, the limit of `C ^ (1/k)` is 1 as `k → ∞`. -/
+lemma tendsto_root_atTop_nhds_one {C : ℝ} (hC : 0 < C) : Filter.Tendsto
+    (fun k : ℕ ↦ C ^ (k : ℝ)⁻¹) Filter.atTop (nhds 1) := by
+  rw [← Real.exp_log hC]
+  simp_rw [← Real.exp_mul]
+  apply Real.tendsto_exp_nhds_zero_nhds_one.comp
+  exact tendsto_const_div_atTop_nhds_zero_nat (Real.log C)
+
+open Filter
+
+/-extends the lemma `tendsto_rpow_div` when the function has natural input-/
+lemma tendsto_nat_rpow_div : Filter.Tendsto (fun k : ℕ ↦ (k : ℝ) ^ (k : ℝ)⁻¹)
+    Filter.atTop (nhds 1) := by
+  simp only [Filter.tendsto_def, Filter.mem_atTop_sets]
+  intro N hN
+  let h := tendsto_rpow_div
+  simp only [Filter.tendsto_def, one_div, Filter.mem_atTop_sets] at h
+  rcases h N hN with ⟨a, ha⟩
+  use (Nat.floor a) + 1
+  intro b hb
+  exact ha b (le_trans (le_of_lt (Nat.lt_floor_add_one a)) (mod_cast hb))
+
+end Filter
+
+namespace Real
+
+/-- `Nat.log` is less than or equal to `Real.log`. -/
+lemma nat_log_le_real_log {a b : ℕ}  (hb : 1 < b) : Nat.log b a ≤ Real.logb b a := by
```

Comment:
`#find_home! nat_log_le_real_log` says that this should go in `Mathlib.Analysis.SpecialFunctions.Log.Base`. Should we move it there or leave it here?

Reply:
I think all of these "Preliminary lemmas" should have better places to go to.